lib/IDE/CodeCompletion.cpp

```diff
@@ -1302,6 +1302,7 @@ class CodeCompletionCallbacksImpl : public CodeCompletionCallbacks {
   TypeLoc ParsedTypeLoc;
   DeclContext *CurDeclContext = nullptr;
   DeclAttrKind AttrKind;
+  SyntaxKind SyntxKind;
```

Comment:
It's a great idea to use syntax kind to differentiate completion scenario👍. However this is unprecedented. Could you add some doc-comments for what this SyntxKind should be used for?